### PR TITLE
Legend property in both line and bar graphs

### PR DESCRIPTION
**Pull request by sridharan-dlt**
### Overview

- Covered all the review points as mentioned in PR' #6 . 
- Modified the code as per review points and test properly from my end.
- Resubmitting the pull request for the legend part.

### CODE REVIEW

File: frontend/apps/monkvision/components/chart-box/lib/chart.mjs [hunk 1]
```diff
@@ -51,15 +51,17 @@ async function drawBargraph(canvas, contents, maxXTicks, gridLines, xAtZero, yAt
...
+    return await _drawLineOrBarGraph(canvas, contents, maxXTicks, gridLines, xAtZero, yAtZeros, ysteps, ylabels, ymaxs, bgColors, brColors, labelColor, gridColor, singleAxis, "line", 0.5, 2, annotation, legend);
 }
 
-async function _drawLineOrBarGraph(canvas, contents, maxXTicks, gridLines, xAtZero, yAtZeros, ysteps, ylabels, ymaxs, bgColors, brColors, labelColor, gridColor, singleAxis, type, pointWidth, lineWidth, annotation) {
+async function _drawLineOrBarGraph(canvas, contents, maxXTicks, gridLines, xAtZero, yAtZeros, ysteps, ylabels, ymaxs, bgColors, brColors, labelColor, gridColor, singleAxis, type, pointWidth, lineWidth, annotation, legend) {
     await _init(); const ctx = canvas.getContext("2d"); 
 
-    const datasets = []; for (const [i,ys] of contents.ys.entries()) datasets.push({ data: ys, 
+    const datasets = []; for (const [i,ys] of contents.ys.entries()) datasets.push({ data: ys, ...contents.legends && { label: contents.legends[i] },     /* Add label property iff legends present in contents */
```
Comment (arpit9616) [inline frontend/apps/monkvision/components/chart-box/lib/chart.mjs:64]: Typo in the comment, please fix.

**Comment by arpit9616** [comment]
Please modify so that legends have sensible text for line graph and bar graph.
Screenshot of current implementation attached below:

image

**Comment by sridharan-dlt** [comment]
@arpit9616 sure, will work on it.

**Comment by arpit9616** [comment]
Looks OK now.

image

**Comment by TekMonksGitHub** [comment]
Merging it.

**Pull request merged**